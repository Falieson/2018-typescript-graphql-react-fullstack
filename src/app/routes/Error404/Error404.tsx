import * as React from 'react'

class Error404 extends React.Component<{
  staticContext: {missed: boolean}
}, {}> {
  componentWillMount() {
    const { staticContext } = this.props
    if (staticContext) staticContext.missed = true
  }

  render() {
    return <div>Sorry, that page was not found.</div>
  }
}

export default Error404
