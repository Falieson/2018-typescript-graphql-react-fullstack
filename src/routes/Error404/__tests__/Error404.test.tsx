/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme'
import * as React from 'react'
import Error404 from '../Error404'

describe('<Error404 />', () => {
  test('renders', () => {
    const staticContext: any = {} // tslint:disable-line no-any
    const wrapper = shallow(<Error404 staticContext={staticContext} />)
    expect(wrapper).toMatchSnapshot()
    expect(staticContext.missed).toBeTruthy()
  })
})
