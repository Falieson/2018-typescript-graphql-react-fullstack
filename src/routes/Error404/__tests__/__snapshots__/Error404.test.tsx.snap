// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error404 /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error404
    staticContext={
      Object {
        "missed": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Sorry, that page was not found.",
    },
    "ref": null,
    "rendered": "Sorry, that page was not found.",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sorry, that page was not found.",
      },
      "ref": null,
      "rendered": "Sorry, that page was not found.",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
